contract-examples/test/fee_manager.ts

```diff
@@ -9,257 +9,87 @@ import {
   ContractFactory,
 } from "ethers"
 import { ethers } from "hardhat"
+const assert = require("assert");
 
 // make sure this is always an admin for the precompile
-const adminAddress: string = "0x8db97C7cEcE249c2b98bDC0226Cc4C2A57BF52FC"
+const ADMIN_ADDRESS: string = "0x8db97C7cEcE249c2b98bDC0226Cc4C2A57BF52FC"
 const FEE_MANAGER = "0x0200000000000000000000000000000000000003";
 
-const ROLES = {
-  NONE: 0,
-  ENABLED: 1,
-  ADMIN: 2
-};
+const GENESIS_CONFIG = require('../../tests/precompile/genesis/fee_manager.json');
 
-const C_FEES = {
-  gasLimit: 8_000_000, // gasLimit
-  targetBlockRate: 2, // targetBlockRate
-  minBaseFee: 25_000_000_000, // minBaseFee
-  targetGas: 15_000_000, // targetGas
-  baseFeeChangeDenominator: 36, // baseFeeChangeDenominator
-  minBlockGasCost: 0, // minBlockGasCost
-  maxBlockGasCost: 1_000_000, // maxBlockGasCost
-  blockGasCostStep: 100_000 // blockGasCostStep
-}
+const testFn = (fnName, overrides = {}) => async function () {
```

Comment:
Can we add a comment here to explain how this is used?

Reply:
Yes, I will do that at the end as I may need to add more functionality with the other tests. I'm going to pull it out and add it in a `utils.ts` file somewhere after.